### PR TITLE
Update async-node-events and event handling.

### PR DESCRIPTION
**Pull request by davidlehn**
- start() now returns a Promise instead of using a callback.
- async-node-events updated.
- Updated for new events API.
- Remove old event object style.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
   "homepage": "https://github.com/digitalbazaar/bedrock",
   "dependencies": {
     "async": "^2.6.0",
```
Comment (mattcollier) [inline package.json]: I guess there's still more async stuff lingering around?

Comment (davidlehn) [reply package.json]: Yes, in logger code.  Not much, but took longer than a few mins to convert so I left it.

Comment (davidlehn) [reply package.json]: Ok, fine, converted it.  Removed async and mkdirp deps and bumped Node.js dep to v10.12.0.

**Comment by davidlehn** [comment]
I should note that I'm not quite sure what should be done with the bedrock.start() api.  The internals went async, so it's now returning a promise, and apps should now catch errors in their top level file.  Shortest version that avoids unhandled exception issues without just dropping the error would be something like:
```js
bedrock.start().catch(err => console.error(err));
```

**Comment by mattcollier** [comment]
> I should note that I'm not quite sure what should be done with the bedrock.start() api. The internals went async, so it's now returning a promise, and apps should now catch errors in their top level file. Shortest version that avoids unhandled exception issues without just dropping the error would be something like:
> 
> ```js
> bedrock.start().catch(err => console.error(err));
> ```

I propose a try/catch or .catch() around start() internally that logs the error here and then rethrow the error.  Something like...
```
bedrock.start().catch(err => {
  console.error(err);
  // is it possible to bedrock logs?
  throw err;
}):
```

**Comment by gannan08** [comment]
> > I should note that I'm not quite sure what should be done with the bedrock.start() api. The internals went async, so it's now returning a promise, and apps should now catch errors in their top level file. Shortest version that avoids unhandled exception issues without just dropping the error would be something like:
> > ```js
> > bedrock.start().catch(err => console.error(err));
> > ```
> 
> I propose a try/catch or .catch() around start() internally that logs the error here and then rethrow the error. Something like...
> 
> ```
> bedrock.start().catch(err => {
>   console.error(err);
>   // is it possible to bedrock logs?
>   throw err;
> }):
> ```

I would do a try/catch here but both achieve the same thing. This looks good to me.

**Comment by davidlehn** [comment]
Re start() promise, it'll just be returned now, and potentially be an error, and later we can add options to handle errors better within `start`:
https://github.com/digitalbazaar/bedrock/issues/56

**Pull request merged**